src/main/java/seedu/address/ui/CommandBox.java

```diff
@@ -1,56 +1,109 @@
 package seedu.address.ui;
 
+import java.util.LinkedList;
+import java.util.ListIterator;
+
 import javafx.collections.ObservableList;
 import javafx.fxml.FXML;
 import javafx.scene.control.TextField;
+import javafx.scene.input.KeyEvent;
 import javafx.scene.layout.Region;
 import seedu.address.logic.commands.CommandResult;
 import seedu.address.logic.commands.exceptions.CommandException;
 import seedu.address.logic.parser.exceptions.ParseException;
 
 /**
- * The UI component that is responsible for receiving user command inputs.
- */
+* The UI component that is responsible for receiving user command inputs.
+*/
 public class CommandBox extends UiPart<Region> {
-
     public static final String ERROR_STYLE_CLASS = "error";
     private static final String FXML = "CommandBox.fxml";
+    private static final int MAX_HISTORY_BUFFER_SIZE = 200;
 
     private final CommandExecutor commandExecutor;
 
     @FXML
     private TextField commandTextField;
 
+    private LinkedList<String> commandHistory = new LinkedList<String>();
+    private ListIterator<String> historyIterator;
```

Comment:
Opening up the application for the first time and immediately pressing UP or DOWN in an attempt to navigate the history results in a `nullPointerException`, since `historyIterator` is not initialised alongside `commandHistory`.

![image](https://user-images.githubusercontent.com/44989315/66403179-b6e3ae00-ea18-11e9-993c-836f0cbfde11.png)

Reply:
fixed